stacks_queues/queue.py

```diff
@@ -15,47 +15,73 @@ def __init__(self):
         self.front = -1
         self.rear = -1
         self.size = 0
-      
+    
 
     def enqueue(self, element):
         """ Adds an element to the Queue
             Raises a QueueFullException if all elements
             In the store are occupied
             returns None
         """
-        pass
+        if self.front == -1:
+            self.front = 0
+            self.rear = 0
+        elif self.size == self.buffer_size:
+            raise QueueFullException('Queue is full')
+        
+        self.store[self.rear] = element
+        self.rear = (self.rear + 1) % self.buffer_size
+        self.size += 1
+
 
     def dequeue(self):
         """ Removes and returns an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
         """
-        pass
+        if self.front == self.rear:
+            raise QueueEmptyException('Queue is empty')
+        
+        front = self.store[self.front]
+        self.store[self.front] = None
+        self.front = (self.front + 1) % self.buffer_size
+        self.size -= 1
+
+        return front
 
     def front(self):
         """ Returns an element from the front
             of the Queue and None if the Queue
             is empty.  Does not remove anything.
         """
-        pass
+        if self.front == self.rear:
+            raise QueueEmptyException('Queue is empty')
+        else:
+            return self.store[self.front]
         
 
     def size(self):
         """ Returns the number of elements in
             The Queue
         """
-        pass
+        return self.size
 
     def empty(self):
```

Comment:
👀 Same thing as your `front` method with `self.front` and `self.rear` being equal. Consider using `self.size` or think about what values `self.front` and `self.rear` will both have if the queue is empty.

However, your overall logic is correct!